pkg/controller/operators/catalog/operator.go

```diff
@@ -1770,3 +1834,54 @@ func getCSVNameSet(plan *v1alpha1.InstallPlan) map[string]struct{} {
 
 	return csvNameSet
 }
+
+func (o *Operator) pollForCRD(pollPeriod time.Duration, crdName string) error {
+	timeout := time.After(pollPeriod)
+	for {
+		select {
+		case <-timeout:
+			return fmt.Errorf("%s CRD not available", crdName)
+		case <-time.After(time.Millisecond * 100):
+			_, err := o.lister.APIExtensionsV1beta1().CustomResourceDefinitionLister().Get(crdName)
+			if err == nil {
+				return nil
+			}
+		}
+	}
+}
+
+func (o *Operator) apiresourceFromGVK(gvk schema.GroupVersionKind) (metav1.APIResource, error) {
+	logger := o.logger.WithFields(logrus.Fields{
+		"group":   gvk.Group,
+		"version": gvk.Version,
+		"kind":    gvk.Kind,
+	})
+
+	resources, err := o.opClient.KubernetesInterface().Discovery().ServerResourcesForGroupVersion(gvk.GroupVersion().String())
+	if err != nil {
+		logger.WithField("err", err).Info("could not query for GVK in api discovery")
+		return metav1.APIResource{}, err
+	}
+	for _, r := range resources.APIResources {
+		if r.Kind == gvk.Kind {
+			return r, nil
+		}
+	}
+	logger.Info("couldn't find GVK in api discovery")
+	return metav1.APIResource{}, olmerrors.GroupVersionKindNotFoundError{gvk.Group, gvk.Version, gvk.Kind}
+}
+
+const (
+	PrometheusRuleKind = "PrometheusRule"
+	ServiceMonitorKind = "ServiceMonitor"
+)
+
+// isSupported returns true if OLM supports this type of CustomResource.
+func isSupported(kind string) bool {
+	switch kind {
```

Comment:
see other note, we could treat our existing objects (role, clusterrole, service, etc) here as well.

Reply:
would be nice if we could tie this with the registry validation work done recently, which basically does the same thing https://github.com/operator-framework/operator-registry/commit/d8e2ec52c19ad2b2a11a91f1ecccb1887bbae787#diff-e2dca81e4297cb851471d15b17acf7f4R23
I know we don't want to depend on the registry in OLM but maybe if we pull these types out and put them in the api long-term it could be nicer. that way there's one place where all our defined types are stored.

Reply:
That's a great suggestion @exdx - we had discussed fleshing out an "operatorframework/apis" repo but did not get terribly far. Let's discuss tomorrow.

I assume that this too ca be closed due to the discussion that @ecordell and I had offline [1]

[1] https://github.com/operator-framework/operator-lifecycle-manager/pull/1227#discussion_r369834013

Reply:
This issue has been discussed on my validation PR on registry as well. See the thread (https://github.com/operator-framework/operator-registry/pull/146#discussion_r366883298). There is a tech debt on consolidating the supported types into some sort of API or importable code that can be imported between different repos. I would say it will need to start with OLM who is the one deciding which resources are supported/consumable. This is something we can discuss and address later as this PR is certainly not a right place to address that issue.